source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
+  ``ClassModelBuilder``.
+
+For more information on how to use the ``@BsonIgnore`` annotation in a POJO,
+see the section of this guide on :ref:`Annotations <annotations>`.
+
+The following sample code shows a custom class that implements the
+``PropertySerialization`` interface to override the default conditions
+by which to determine whether to serialize a field:
 
-- Description
-  - Define term: discriminator
-- Example
-  - Show each way (annotation and ClassModelBuilder method) to enable discriminator
-  
+.. code-block:: java
+   :emphasize-lines: 4
 
-Supporting POJOs without Default Constructors
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   public class CourteousAgeSerialization implements PropertySerialization<Integer> {
+       @Override
+       public boolean shouldSerialize(Integer value) {
+           return (value < 30);
+       }
+   }
 
-- Description, explaining when this is used
-- Example of BsonCreator and BsonProperty annotations to specify the constructor and parameter names to use
+The class above specifies that any integer above 29 is not serialized and
+therefore, not included in the MongoDB document. Suppose you applied this
+custom serialization behavior to the following sample POJO:
 
-Customizing Serialization
-~~~~~~~~~~~~~~~~~~~~~~~~~
+.. code-block:: java
 
-- Description, when to use
-- Examples
-  - Annotation
-  - PropertySerialization in PropertyModelBuilder
+   public class BirthdayInvitation {
+       private String name;
+       private Integer age;
+       private LocalDateTime eventDateTime;
 
+       // ...
+   }
+
+You can specify the custom serialization by adding the
+``CourteousAgeSerialization`` instance to the ``PropertyModelBuilder`` from
+the ``ClassModel`` property associated with the ``age`` field using the
+following code:
+
+.. code-block:: java
+
+   ClassModelBuilder<BirthdayInvitation> classModel = ClassModel.builder(BirthdayInvitation.class);
+   ((PropertyModelBuilder<Integer>) classModel.getProperty("age"))
+           .propertySerialization(new CourteousAgeSerialization());
+
+   PojoCodecProvider pojoCodecProvider = PojoCodecProvider.builder().register(classModel.build()).build();
+   CodecRegistry pojoCodecRegistry = fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+If you insert a POJO that contains a value greater than 29 in the ``age``
+field, the serialized document omits it. The resulting document should
+resemble the following:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id" : ObjectId("..."), "eventDateTime" : ..., "name" : "Galadriel" }
 
 Generics Support
 ^^^^^^^^^^^^^^^^
 
-- Description of when a POJO is not serializable
-- Example showing how to use PropertyCodecProviders to provide concrete class information
+You can use the POJO ``Codec`` to serialize classes that contain generic
+properties if they meet the following criteria:
+
+- contain only bounded concrete type parameters
```

Comment:
I would suggest adding a subject like "the top-level POJO and all subclasses must" for consistency with the next bullet point

Reply:
Additionally, a link to what "concrete type parameters" (or an explanation below) would help this be more understandable to readers.

Reply:
Will update the bullet point and I'll add a link to the Oracle docs for info on type parameters at the end of the section.